src/Chapter3.hs

```diff
@@ -910,6 +1067,24 @@ Implement instances of "Append" for the following types:
 class Append a where
     append :: a -> a -> a
 
+newtype Gold = Gold { value :: Int}
+
+instance Append Gold where
+  append :: Gold -> Gold -> Gold
+  append g1 g2 = Gold((value g1) + (value g2))
+
+instance Append [a] where
+  append :: [a] -> [a] -> [a]
+  append l1 l2 = l1 ++ l2
+
+-- instance Append (Maybe x) where
+--   append :: (Maybe x) -> (Maybe x) -> (Maybe x)
+--   append (Just x) (Just y) = append x y
+--   append (Just x) Nothing = Just x
+--   append Nothing (Just y) = Just y
+--   append Nothing Nothing = Nothing
```

Comment:
This is an absolutely correct solution!

However, we can notice that these 3 cases could be reduced to 2!
Let's see what we can unite in here.

If some of the arguments are `Nothing`, you actually always return the second one! So you need to inspect only two cases: `Nothing` in the first position, and `Nothing` on the second one 🙂

Reply:
oh yes, thanks for pointing this out !